hyppo/conditional_independence/base.py

```diff
@@ -0,0 +1,136 @@
+from abc import ABC, abstractmethod
+from typing import NamedTuple
+
+from ..tools import perm_test
+
+
+class IndependenceTestOutput(NamedTuple):
+    stat: float
+    pvalue: float
+
+
+class IndependenceTest(ABC):
```

Comment:
This should be called condtional independence test or something like in Matt's PR